lms/djangoapps/courseware/entrance_exams.py

```diff
@@ -92,12 +92,13 @@ def _calculate_entrance_exam_score(user, course_descriptor, exam_modules):
     exam_module_ids = [exam_module.location for exam_module in exam_modules]
 
     # All of the corresponding student module records
-    student_modules = StudentModule.objects.filter(
+    student_module_dict = {}
+    student_modules = StudentModule.objects.chunked_filter(
+        'module_state_key__in',
+        exam_module_ids,
         student=user,
-        course_id=course_descriptor.id,
-        module_state_key__in=exam_module_ids,
+        course_id=course_descriptor.id
     )
```

Comment:
We've been centralizing `StudentModule` access inside the `XBlockUserStateClient` interface (implemented in edx-platform here: https://github.com/edx/edx-platform/blob/master/lms/djangoapps/courseware/user_state_client.py#L22). We're centralizing that access because we're in the process of evaluating replacement backing stores for this data. Can you please use the `get_many` method here, rather than querying `StudentModule` directly?